### DISCUSSION TITLE
Question about using qpdf decrypt command

### DISCUSSION THREAD

**Discussion by alokdharia** [qpdf/qpdf#735] [General]
I am doing a migration of PDF files. We have a good amount of  PDF files which are secured with a password. We want to load these files to target repository without the password (unsecured). I have the password with me and I plan to use following command.

qpdf.exe --decrypt --verbose --password=test1234 input.pdf  output1.pdf

I am not sure as how this command is executed, but most likely it seems that it creates a new output file without the password.
I want to ensure that when the new file is created, there is no change in the resolution (Jpeg image etc) or change in fonts. I am also not sure as how it would behave if a font is missing from the server where I am running from.

Can some one help me to understand the risks, if I were to use this command. (example will i see poor resolution, different fonts etc)

**Comment by jberkenbilt** [comment]
Short answer: this is an extremely low-risk operation. You will definitely not see changes in image resolution or fonts. Detailed answer below.

All this does is write the file out with no encryption. It won't change the content in any way. If qpdf exits with a status of zero, you can be quite confident that the file is intact and undamaged. qpdf is not interpreting the contents of the PDF. Removing encryption is basically a structural change. Any errors like missing fonts, corrupted images, etc., would carry across from the encrypted to the unencrypted file.

If there were structural problems with the PDF, that could cause qpdf to have problems, but in that case, qpdf would give errors or warnings. That said, it is very rare for encrypted files to have these kinds of problems. This is for two reasons: one is that errors in encrypted files tend to make the file completely unusable (so the source file would be broken), and the other one is that most broken PDFs come from broken PDF writers. Encryption is tricky, and in 20+ years of dealing with millions of PDF files (no exaggeration), I have yet to see a PDF writer that creates structurally broken PDF files and that is also capable of creating encrypted files.

Reminds me of a joke about someone who goes to the record store (old joke) to by a Mozart CD and wants to pay with a check. The cashier lets her buy the CD without verifying the check. She says, "Don't you need to check my ID or something?" The clerk responds, "Nah, people who buy Mozart don't write bad checks."

**Reply by SH2282000** [reply]
Also, to complete the answer of @jberkenbilt: with **multiple** encrypted pdf with the same password in a directory and its sub-directories, the following script could help you to save time by unlocking them all at once: 

```
for eachfile in *.pdf
do
   echo $eachfile
    qpdf --decrypt --password=PASSWORD --replace-input $eachfile
done
```

- [ ] Copy the directory before hand
- [ ] Change PASSWORD
- [ ] input the script in the cmd line while being in the copied directory

**Comment by alokdharia** [comment]
Thank you Jay and Shannah for your quick response. This helps me.
